### PR TITLE
dxLookup - Popup position is not correctly defined in the Material theme

### PR DESCRIPTION
**Pull request by groshenkovamarina**

### CODE REVIEW

File: js/ui/lookup.js [hunk 1]
```diff
@@ -489,36 +489,65 @@ const Lookup = DropDownList.inherit({
...
+        const listItemsCount = this._list.option('items').length;
+        const lookupOffset = $(this._list.element()).offset().top;
 
-        const selectedIndex = this._list.option('selectedIndex');
-        const flipped = this._popup._$wrapper.hasClass(LOOKUP_POPOVER_FLIP_VERTICAL_CLASS);
-        if(selectedIndex === -1 || flipped) return;
+        let offset = 0;
 
-        const selectedListItem = $(this._list.element()).find('.' + LIST_ITEM_SELECTED_CLASS);
+        if(1 < selectedIndex && selectedIndex < (listItemsCount - 2)) {
```
Comment (babich-a) [inline js/ui/lookup.js]: I think we need to add some info about what this condition means. It is unclear now. Is it two 'border' items at the start and the end of the list?

Comment (groshenkovamarina) [reply js/ui/lookup.js]: Yes, it is two items at the start and the end of the list, which cannot be centered. What do you suggest? Add comment?

Comment (babich-a) [reply js/ui/lookup.js]: I prefer the self-documenting code. We can extract this to the function for example

Comment (groshenkovamarina) [reply js/ui/lookup.js]: We can extract internal code from 'If' condition, but it will not more clear

Comment (babich-a) [reply js/ui/lookup.js]: it depends on function name :)

Comment (groshenkovamarina) [reply js/ui/lookup.js]: What do you think about 'getCenteringOffset' name?

Comment (babich-a) [reply js/ui/lookup.js]: needCentering(){ return selectedIndex > 1 && selectedIndex < (listItemsCount - 2);}
smth like this?

Comment (groshenkovamarina) [reply js/ui/lookup.js]: Updated

File: js/ui/lookup.js [hunk 2]
```diff
@@ -489,47 +489,99 @@ const Lookup = DropDownList.inherit({
...
+
+        if(selectedIndex !== -1) {
+            if(this._isCenteringEnabled(selectedIndex, listItemsCount)) {
+                this._scrollToSelectedItem();
+
+                const scrollOffsetTop = (popupHeight - selectedListItem.height()) / 2 - this._getDifferenceOffsets(selectedListItem);
+
+                this._list.scrollTo(this._list.scrollTop() + MATERIAL_LOOKUP_LIST_PADDING / 2 - scrollOffsetTop);
+
+                offsetTop = differenceOfHeights + this._getDifferenceOffsets(selectedListItem);
```
Comment (babich-a) [inline js/ui/lookup.js:525]: It seems that this line is need in all condition branches (522, 527, 532, 536)

Comment (groshenkovamarina) [reply js/ui/lookup.js:525]: Yes, but  this._getDifferenceOffsets(selectedListItem) need calculate after scroll each time

Comment (babich-a) [reply js/ui/lookup.js:525]: And? Can we calculate it once after all scrollings? Duplicate code 4 times is not good solution

Comment (groshenkovamarina) [reply js/ui/lookup.js:525]: As you can see in the first case this._getDifferenceOffsets(selectedListItem) calls twice. The second time is based on the previous value of this._getDifferenceOffsets(selectedListItem)

Comment (groshenkovamarina) [reply js/ui/lookup.js:522]: I can extract this line from the other two cases

**Review by babich-a — DISMISSED** [review]

**Review by babich-a — DISMISSED** [review]

**Pull request closed without merging**